src/main/java/io/xpring/xrpl/model/XRPCurrencyAmount.java

```diff
@@ -0,0 +1,56 @@
+package io.xpring.xrpl.model;
+
+import org.xrpl.rpc.v1.CurrencyAmount;
+import org.immutables.value.Value;
+import org.xrpl.rpc.v1.IssuedCurrencyAmount;
+
+import javax.annotation.Nullable;
+
+/**
+ * An amount of currency on the XRP Ledger
+ * @see "https://xrpl.org/basic-data-types.html#specifying-currency-amounts"
+ */
+@Value.Immutable
+public interface XRPCurrencyAmount {
+    static ImmutableXRPCurrencyAmount.Builder builder() {
+        return ImmutableXRPCurrencyAmount.builder();
+    }
+
+    /**
+     * @return An amount of XRP, specified in drops.
+     * Mutually exclusive fields - only drops XOR issuedCurrency should be set.
+     */
+     @Nullable String drops();
+
+    /**
+     * @return An amount of an issued currency.
+     * Mutually exclusive fields - only drops XOR issuedCurrency should be set.
+     */
+    @Nullable XRPIssuedCurrency issuedCurrency();
+
+    static XRPCurrencyAmount from(CurrencyAmount currencyAmount) {
+        switch (currencyAmount.getAmountCase()) {
+            // Mutually exclusive: either drops or issuedCurrency is set in an XRPCurrencyAmount
+            case ISSUED_CURRENCY_AMOUNT: {
+                IssuedCurrencyAmount issuedCurrencyAmount = currencyAmount.getIssuedCurrencyAmount();
+                if (issuedCurrencyAmount!= null) {
```

Comment:
I believe in java you can drop the `!= null` if you want and just write:
```
if (issuedCurrency) {
```

Java evaluated objects that are non-null to be truthy values and objects that are null to be falsy

Reply:
It actually won't let me...  I get a `Required Type: boolean, Provided Type: XRPIssuedCurrency` error.  Java just likes to hear itself talk.

Reply:
Yeah that would be nice...